internal/provider/data_source_outpost_sc_kubernetes.go

```diff
@@ -0,0 +1,68 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataOutpostServiceConnectionsKubernetes() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataOutpostServiceConnectionsKubernetesRead,
+		Description: "Applications --- Get Service Connections for Kubernetes",
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"kubeconfig": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Optional:  true,
+				Sensitive: true,
+			},
+			"local": {
+				Type:     schema.TypeBool,
+				Computed: true,
+				Optional: true,
+			},
+			"verify_ssl": {
+				Type:     schema.TypeBool,
+				Computed: true,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataOutpostServiceConnectionsKubernetesRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	c := m.(*APIClient)
+
+	req := c.client.OutpostsApi.OutpostsServiceConnectionsKubernetesList(ctx)
+	if s, ok := d.GetOk("name"); ok {
+		req = req.Name(s.(string))
```

Comment:
can remove the if here since `name` is marked as required and set that on the request inline